src/librustc_mir_build/build/mod.rs

```diff
@@ -790,12 +791,22 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         argument_scope: region::Scope,
         ast_body: &'tcx hir::Expr<'tcx>,
     ) -> BlockAnd<()> {
+        let tcx = self.hir.tcx();
+        let attrs = tcx.codegen_fn_attrs(fn_def_id);
+        let naked = attrs.flags.contains(CodegenFnAttrFlags::NAKED);
+
         // Allocate locals for the function arguments
         for &ArgInfo(ty, _, arg_opt, _) in arguments.iter() {
             let source_info =
                 SourceInfo::outermost(arg_opt.map_or(self.fn_span, |arg| arg.pat.span));
             let arg_local = self.local_decls.push(LocalDecl::with_source_info(ty, source_info));
 
+            // Emit function argument debuginfo only for non-naked functions.
+            // See: https://github.com/rust-lang/rust/issues/42779
+            if naked {
+                continue;
+            }
```

Comment:
I don't think this is a good idea, wouldn't be safer to just disable *all* debuginfo generation for `#[naked]` functions in codegen?

Reply:
Also this PR doesn't add a `mir-opt` test showing the lack of MIR debuginfo (but which we might not want anyway).

Reply:
@eddyb We don't want to suppress the debuginfo describing the calling contract of the function, just the internal function operations. The lldb can reference this information to show the contents of (for example) the first argument in `%rdi`. But if you think we need to implement this another way, I'm not particular.